core/src/main/java/org/apache/iceberg/ManifestWriter.java

```diff
@@ -178,8 +178,10 @@ public ManifestFile toManifestFile() {
 
   @Override
   public void close() throws IOException {
-    this.closed = true;
-    writer.close();
+    if (writer != null) {
+      writer.close();
+      this.closed = true;
```

Comment:
Nit: This should probably still set `this.closed = true` even if the writer is null.